Adds Learn CSS Sizing (#5077)
GoogleChrome/web.dev: src/site/content/en/learn/css/sizing/index.md

      </tr>
    </tbody>
  </table>
</div>

```css
div {
	width: 10vw;
}

p {
	max-width: 60ch;
}
```
<figure class="w-figure">
{% Codepen {
  user: 'web-dot-dev',
  id: 'JjEYqXa'
} %}
</figure>

In this example, the `div` will be 10% of the viewport's width because `1vw` is **1% of the viewport width**.
The `p` element has a `max-width` of `60ch`
which means it can't exceed the width of 60 "0" characters in the calculated font and size.

{% Aside 'objective' %}

By sizing text with relative units like `em` or `rem`,
rather than an absolute unit, like `px`, the size of your text can respond to user preferences.
This can include the system font size or parent element's font size, such as the `<body>`.
The base size of the `em` is the element's parent and the base size of the `rem` is the base font size of the document.

If you don't define a `font-size` on your `html` element,
this user-preferred system font size will be honoured if you use relative lengths,
such as `em` and `rem`.
If you use `px` units for sizing text,
this preference will be ignored.

{% Video src="video/VbAJIREinuYvovrBzzvEyZOpw5w1/cUEl77VN4ZtAdElV4fd6.mp4" %}
{% endAside %}

> ## Miscellaneous units

Reviewer: maybe add some subheaders for these misc units? it feels like it trails off and thins out, and could be fixed with some clear headers?

Author: Agreed, I've just pushed a little change to that section. @andy-piccalilli maybe take a quick look?

---
title: Sizing Units
description:
authors:
  - andybell
date: 2021-04-13
tags:
  - css
---

# Sizing

> The responsive web is a flexible medium,

Reviewer 2: ```suggestion
The web is a responsive medium,
```

Feel free to ignore, just felt a little repetitive bw the first and last sentence "flexible medium".

```css
div {
	width: 450px;
	height: 100px;
}

div p {
	width: 50%; /* calculated: 225px */
	transform: translateX(10%); /* calculated: 22.5px */
}
```
<figure class="w-figure">
{% Codepen {
  user: 'web-dot-dev',
  id: 'xxgwBxv'
} %}
</figure>

If you set a `transform` value as a percentage,
it is based on the element with the transform set.
In this example, the `p` has a `translateX` value of `10%` and a `width` of `50%`.
First, calculate what the width will be: `225px` because it is **50% of its parent's width**.
Then, take `10%` of `225px`, which is `22.5px`.

{% Aside 'key-term' %}
The transform property allows you alter an element's appearance and position by rotating, skewing, scaling and translating it.
This can be done in a 2D and 3D space.
{% endAside %}

## Dimensions and lengths

If you attach a unit to a number, it becomes a dimension.
For example, `1rem` is a dimension.
In this context, the unit that is attached to a number is referred to in specifications as a dimension token.
Lengths are **dimensions that refer to distance** and they can either be absolute or relative.

### Absolute lengths

All absolute lengths resolve against the same base,

> this makes them predictable wherever they're used in your CSS.

Reviewer 2: ```suggestion
making them predictable wherever they're used in your CSS.
```

        <td>1% of the viewport's larger dimension.</td>
      </tr>
    </tbody>
  </table>
</div>

```css
div {
	width: 10vw;
}

p {
	max-width: 60ch;
}
```
<figure class="w-figure">
{% Codepen {
  user: 'web-dot-dev',
  id: 'JjEYqXa'
} %}
</figure>

In this example, the `div` will be 10% of the viewport's width because `1vw` is **1% of the viewport width**.
The `p` element has a `max-width` of `60ch`
which means it can't exceed the width of 60 "0" characters in the calculated font and size.

{% Aside 'objective' %}

By sizing text with relative units like `em` or `rem`,
rather than an absolute unit, like `px`, the size of your text can respond to user preferences.
This can include the system font size or parent element's font size, such as the `<body>`.
The base size of the `em` is the element's parent and the base size of the `rem` is the base font size of the document.

If you don't define a `font-size` on your `html` element,
this user-preferred system font size will be honoured if you use relative lengths,
such as `em` and `rem`.
If you use `px` units for sizing text,
this preference will be ignored.

> {% Video src="video/VbAJIREinuYvovrBzzvEyZOpw5w1/cUEl77VN4ZtAdElV4fd6.mp4" %}

Reviewer 2: This video doesn't seem to be playing for me 🤔

Author: @andy-piccalilli video isn't playing for me either ... any ideas?

Reviewer 3: I'll tag in @robdodson because the video renders no `controls` or `autoplay` by default, which I imagine we can pass as shortcode params?

Reviewer 4: following up from chat. I think you can configure the player using these controls (https://web.dev/handbook/markup-media/#video-properties-%28videoargs%29) but if the video is still not playing perhaps the compression or something is off? Let me know if I can help.

{% endAside %}

## Miscellaneous units

There are some other units which have been specified to deal with particular types of values.

### Angle units

In the [color module](/learn/css/color/),
we looked at **angle units**,
which are helpful for defining degree values,
such as the hue in `hsl`.
They are also useful for rotating elements within transform functions.

```css
div {
	width: 150px;
	height: 150px;
	transform: rotate(60deg);
}
```
<figure class="w-figure">
{% Codepen {
  user: 'web-dot-dev',
  id: 'VwPvRbK'
} %}
</figure>

Using the `deg` angle unit, you can rotate a `div` 90° on its center axis.

> 

Reviewer 2: ```suggestion
```suggestion
Other angle units include `rad` (radians), `grad` (gradians), and `turn` units, which represent a part of an angle, where `1turn` = `360deg`, and `0.5turn` = `180deg`. These can come in handy when styling transforms and animations.
```
```

> ```css

Reviewer 2: I think this example is in the wrong subhead, it should be moved inside/under the next header (Resolution Units)
